distribution/tools/keystore-cli/src/main/java/org/elasticsearch/common/settings/AddStringKeyStoreCommand.java

```diff
@@ -69,8 +60,7 @@ protected void executeCommand(Terminal terminal, OptionSet options, Environment
 
         final char[] value;
         if (options.has(stdinOption)) {
-            BufferedReader stdinReader = new BufferedReader(new InputStreamReader(getStdin(), StandardCharsets.UTF_8));
-            value = stdinReader.readLine().toCharArray();
+            value = terminal.readSecret("");
```

Comment:
I agree that this can be improved further; however, the original code was failing. In the case when both the password and the string keystore setting were provided over standard input, e.g. `echo "password\nbar\n" | bin/elasticsearch-keystore add foo`, we were using one `BufferedReader` to read the password and another one to read the key value. The problem was that the first `BufferedReader` was slurping both values from standard input into its buffer and leaving nothing for the second `BufferedReader` to read.

The change I made delegated everything to the `Terminal` class, which in the non-interactive case provided a single `BufferedReader` for reading the keystore password and reading the new value.

Perhaps a better solution would be to set the password-reading `BufferedReader` to have a buffer of size zero so that it never reads too much input. In any case, I will figure out how to avoid delegating to the terminal here.